web/src/main/java/org/springframework/security/web/util/UrlUtils.java

```diff
@@ -61,7 +65,7 @@ public static String buildFullRequestUrl(String scheme, String serverName,
 		}
 		else if ("https".equals(scheme)) {
 			if (serverPort != 443) {
-				url.append(":").append(serverPort);
```

Comment:
I don't think this is worth the inconsistency in the code base. If it were, then I'm sure the `append(String)` method could call `append(char)` if it were a single character.

Can you please revert the changes to the append? At that point I we can merge the fix for `Pattern`